File: Recursion/subsequence_pattern/all_subsequences_of_a_string.py
def helper(idx, s, ans, ds):
    if idx == len(s):
        ans.append(ds)
        return

    ds += s[idx]
    helper(idx + 1, s, ans, ds)
    ds = ds[:-1]
    helper(idx + 1, s, ans, ds)


def power_set(s):
    ans = []
    ds = ""
    helper(0, s, ans, ds)
    ans.sort()
    return ans

File: Recursion/subsequence_pattern/AllSubsequencesOfAString.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class AllSubsequencesOfAString {
    public static List<String> powerSet(String s) {
        List<String> ans = new ArrayList<>();
        generate(s, 0, "", ans);
        Collections.sort(ans);
        return ans;
    }

    private static void generate(String s, int idx, String ds, List<String> ans) {
        if (idx == s.length()) {
            ans.add(ds);
            return;
        }

        generate(s, idx + 1, ds + s.charAt(idx), ans);
        generate(s, idx + 1, ds, ans);
    }
}